src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```
@@ -1357,6 +1373,42 @@ export default class Fms {
     }
 
     /**
+     * Regenerate SID leg based on current runway assignment
+     *
+     * @for Fms
+     * @method _regenerateSidLeg
+     */
+    _regenerateSidLeg() {
+        const sidLegIndex = this._findLegIndexForProcedureType(PROCEDURE_TYPE.SID);
+
+        if (sidLegIndex === INVALID_INDEX) {
+            return;
```

If that's the case we should probably throw here instead of just returning.